src/meta/model/migration/src/m20250819_014448_add_version_column_indices_to_table.rs

```diff
@@ -0,0 +1,121 @@
+use sea_orm::{FromJsonQueryResult, FromQueryResult, Statement};
+use sea_orm_migration::prelude::{Table as MigrationTable, *};
+use serde::{Deserialize, Serialize};
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        // Add the new version_column_indices column
+        manager
+            .alter_table(
+                MigrationTable::alter()
+                    .table(Table::Table)
+                    .add_column(
+                        ColumnDef::new(Table::VersionColumnIndices)
+                            .json_binary()
+                            .not_null(),
+                    )
+                    .to_owned(),
+            )
+            .await?;
+
+        // Migrate data from version_column_index to version_column_indices using I32Array serialization
+        let connection = manager.get_connection();
+        let database_backend = connection.get_database_backend();
+
+        // First, get all tables that have a version_column_index
+        let (sql, values) = Query::select()
+            .columns([Table::TableId, Table::VersionColumnIndex])
+            .from(Table::Table)
+            .and_where(Expr::col(Table::VersionColumnIndex).is_not_null())
+            .build_any(&*database_backend.get_query_builder());
+
+        let query_result = connection
+            .query_all(Statement::from_sql_and_values(
+                database_backend,
+                sql,
+                values,
+            ))
+            .await?;
+
+        // For each table with a version_column_index, create the proper I32Array JSON
+        for row in query_result {
+            let table_entity = TableEntity::from_query_result(&row, "")?;
+
+            // Create I32Array with single element
+            let i32_array = I32Array(vec![table_entity.version_column_index.unwrap()]);
+
+            // Update with the I32Array
+            manager
+                .exec_stmt(
+                    Query::update()
+                        .table(Table::Table)
+                        .value(Table::VersionColumnIndices, i32_array)
+                        .and_where(Expr::col(Table::TableId).eq(table_entity.table_id))
+                        .to_owned(),
+                )
+                .await?;
+        }
```

Comment:
I can't seems find a way to use a `set` query to set the new column to json_array(old_value) @yezizp2012